File: .github/benchmarks/run-benchmark.py
#!/usr/bin/env python3
"""
Execute the savage benchmark analysis on selected commands
"""

import sys
import os
sys.path.append('.github/benchmarks/tools')

from benchmark_engine import SavageCommandBenchmarker
import json

def main():
    # Our randomly selected commands for scientific roasting
    selected_commands = [
        "ultrathink-hybrid-mcp.md",
        "adaptive-complexity-router.md", 
        "adhd-context-switch.md",
        "intelligent-code-enhancer.md",
        "intelligent-refactor-session.md"
    ]
    
    print("🔬 SAVAGE COMMAND BENCHMARKER")
    print("=" * 50)
    print("Scientific Analysis of Command Quality with PhD-level Brutality")
    print()
    print(f"Selected commands for roasting: {len(selected_commands)}")
    for i, cmd in enumerate(selected_commands, 1):
        print(f"  {i}. {cmd}")
    print()
    
    # Initialize the roaster
    benchmarker = SavageCommandBenchmarker()
    
    # Run comprehensive benchmarks
    print("🔥 Commencing Scientific Roasting...")
    print()
    
    results = benchmarker.run_full_benchmark(selected_commands)
    
    print()
    print("📊 BENCHMARK RESULTS")
    print("=" * 50)
    
    # Quick summary
    for command, result in results.items():
        print(f"📍 {command}: {result.savage_score:.1f}/10")
        print(f"   Complexity: {result.metrics.complexity_score:.1f}/5")
        print(f"   Success Rate: {result.metrics.success_rate:.1%}")
        print(f"   Token Usage: {result.metrics.token_consumption_total:,}")
        print()
    
    # Generate final report
    print("📝 Generating comprehensive savage report...")
    report_file = benchmarker.generate_final_report(results)
    
    print(f"✅ Complete report saved to: {report_file}")
    print()
    print("🔍 Key Findings:")
    
    # Load and show key insights
    with open(report_file, 'r') as f:
        report_data = json.load(f)
    
    print()
    print("💀 SAVAGE INSIGHTS:")
    for insight in report_data['savage_insights']:
        print(f"  • {insight}")
    
    print()
    print("🧪 SCIENTIFIC CONCLUSIONS:")
    for conclusion in report_data['scientific_conclusions']:
        print(f"  • {conclusion}")
    
    print()
    print("🏆 COMMAND RANKINGS:")
    for ranking in report_data['command_rankings'][:3]:  # Top 3
        print(f"  {ranking['rank']}. {ranking['command']}: {ranking['savage_score']:.1f}/10")
    
    print()
    print(f"📄 Full report with statistical analysis: {report_file}")
    print("💀 The roasting is complete. Your commands have been scientifically judged.")

if __name__ == "__main__":
    main()

File: .github/benchmarks/run-benchmark.js
#!/usr/bin/env node

const { SavageCommandBenchmarker } = require('./benchmark-harness.js');
const fs = require('fs');
const path = require('path');

async function main() {
    console.log(`
╔════════════════════════════════════════════════════════════════════════════════╗
║                                                                                ║
║         🔬 SAVAGE COMMAND BENCHMARKER - Scientific Roasting Session           ║
║                                                                                ║
║    "Any fool can write code that a computer can understand.                    ║
║     Good programmers write code that humans can understand." - Martin Fowler  ║
║                                                                                ║
╚════════════════════════════════════════════════════════════════════════════════╝
`);

    const benchmarker = new SavageCommandBenchmarker();
    
    try {
        console.log("📊 Executing comprehensive benchmark suite...\n");
        
        const results = await benchmarker.runCompleteBenchmark();
        
        // Generate timestamp for report
        const timestamp = new Date().toISOString().replace(/[:.]/g, '-').substring(0, 19);
        const reportPath = path.join(__dirname, 'results', `${timestamp}-report.json`);
        
        // Ensure results directory exists
        const resultsDir = path.dirname(reportPath);
        if (!fs.existsSync(resultsDir)) {
            fs.mkdirSync(resultsDir, { recursive: true });
        }
        
        // Save detailed JSON report
        fs.writeFileSync(reportPath, JSON.stringify(results, null, 2));
        
        console.log(`📋 Detailed results saved to: ${reportPath}\n`);
        
        // Generate human-readable summary
        console.log("🎯 EXECUTIVE SUMMARY - THE BRUTAL TRUTH:");
        console.log("=" .repeat(80));
        
        Object.entries(results.commands).forEach(([commandName, data]) => {
            console.log(`\n📊 ${commandName.toUpperCase()}`);
            console.log("-".repeat(50));
            console.log(`📈 Token Consumption: ${data.metrics.tokenConsumption.statistics.mean.toFixed(0)} ± ${data.metrics.tokenConsumption.statistics.stdDev.toFixed(0)}`);
            console.log(`⏱️  Execution Time: ${(data.metrics.executionTime.statistics.mean / 1000).toFixed(2)}s ± ${(data.metrics.executionTime.statistics.stdDev / 1000).toFixed(2)}s`);
            console.log(`✅ Success Rate: ${data.metrics.successRate.successRate.toFixed(1)}%`);
            console.log(`🔢 Complexity Score: ${data.metrics.complexityScore.score} (${data.metrics.complexityScore.verdict})`);
            console.log(`🏆 Overall Verdict: ${data.verdict}`);
            
            if (data.savageComments.length > 0) {
                console.log("\n💀 SAVAGE BUT FAIR ASSESSMENT:");
                data.savageComments.forEach(comment => {
                    console.log(`   💣 ${comment}`);
                });
            }
        });
        
        console.log("\n" + "=".repeat(80));
        console.log("🎓 PROFESSOR'S FINAL GRADES:");
        console.log("=".repeat(80));
        
        const grades = Object.entries(results.commands).map(([name, data]) => ({
            name,
            verdict: data.verdict,
            complexity: data.metrics.complexityScore.score,
            success: data.metrics.successRate.successRate
        })).sort((a, b) => b.success - a.success);
        
        grades.forEach((cmd, index) => {
            const medal = index === 0 ? "🥇" : index === 1 ? "🥈" : index === 2 ? "🥉" : "💩";
            console.log(`${medal} ${cmd.name}: ${cmd.verdict} (Success: ${cmd.success.toFixed(1)}%, Complexity: ${cmd.complexity})`);
        });
        
        console.log(`\n📊 Report timestamp: ${timestamp}`);
        console.log(`📁 Full data: ${reportPath}`);
        console.log("\n🎯 Benchmarking complete. Science has spoken.");
        
    } catch (error) {
        console.error("💥 Benchmark failed:", error);
        process.exit(1);
    }
}

if (require.main === module) {
    main().catch(console.error);
}

module.exports = { main };